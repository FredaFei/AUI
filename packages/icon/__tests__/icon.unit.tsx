import * as renderer from 'react-test-renderer'
import React from 'react'
import {shallow,mount} from 'enzyme'

import Icon from '../icon'

describe('测试 Icon 以下行为', () => {
  it('Icon 存在', () => {
    const json = renderer.create(<Icon name="settings"/>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('Icon 设置 rotate', () => {
    const wrapper = mount(<Icon name="settings" rotate={30}/>)
    expect(wrapper.props().rotate).toBe(30)
  })
  it('Icon 设置 spin', () => {
    const wrapper = shallow(<Icon name="settings" spin/>)
    expect(wrapper.find('svg').hasClass('am-icon-spin')).toBe(true)
  })
  it('Icon 设置 loading', () => {
    const wrapper = shallow(<Icon name="loading"/>)
    expect(wrapper.find('svg').hasClass('am-icon-spin')).toBe(true)
  })
  it('Icon 设置 onClick', () => {
    const mockCallBack = jest.fn();
    const wrapper = shallow(<Icon name="settings" onClick={mockCallBack}/>);
    wrapper.simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
  it('Icon 设置 自定义类名', () => {
    const wrapper = shallow(<Icon name="left" className="test" />)
    expect(wrapper.find('svg').hasClass('test')).toBe(true)
  })
  it('Icon 设置 自定义样式', () => {
    const wrapper = shallow(<Icon name="left" style={{color: 'red'}} />)
    expect(wrapper.render()).toMatchSnapshot()
  });
})