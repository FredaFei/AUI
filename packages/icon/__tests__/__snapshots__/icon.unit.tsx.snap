// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 Icon 以下行为 Icon 存在 1`] = `
<svg
  className="am-icon "
  style={Object {}}
>
  <use
    xlinkHref="#settings"
  />
</svg>
`;

exports[`测试 Icon 以下行为 Icon 设置 自定义样式 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "am-icon ",
      "style": "color:red",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "href": "#left",
        },
        "children": Array [],
        "name": "use",
        "namespace": "http://www.w3.org/2000/svg",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "href": "http://www.w3.org/1999/xlink",
        },
        "x-attribsPrefix": Object {
          "href": "xlink",
        },
      },
    ],
    "name": "svg",
    "namespace": "http://www.w3.org/2000/svg",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
